streams/src/test/java/org/apache/kafka/streams/state/StreamsMetadataTest.java

```diff
@@ -55,6 +62,63 @@ public void shouldNotAllowModificationOfInternalStateViaGetters() {
         assertTrue(isUnmodifiable(streamsMetadata.standbyStateStoreNames()));
     }
 
+    @Test
+    public void shouldFollowHashCodeAndEqualsContract() {
```

Comment:
@cadonna For hash code and equals contract validation, I decided to keep it under one single test, and validate the positive case and any of the reasons why it might fall into the negative case.

Reply:
see my other comment about separate tests.